aleph/views/diagrams_api.py

```diff
@@ -0,0 +1,178 @@
+import logging
+from banal import first
+from flask import Blueprint, request
+
+from aleph.model import Diagram
+from aleph.search import QueryParser, DatabaseQueryResult
+from aleph.views.serializers import DiagramSerializer
+from aleph.views.util import get_db_collection, parse_request
+from aleph.views.util import obj_or_404, require
+
+
+blueprint = Blueprint('diagrams_api', __name__)
+log = logging.getLogger(__name__)
+
+
+@blueprint.route('/api/2/diagrams', methods=['GET'])
+def index():
+    """Returns a list of diagrams for the role
+    ---
+    get:
+      summary: List diagrams
+      parameters:
+      - description: The collection id.
+        in: query
+        name: 'filter:collection_id'
+        required: true
+        schema:
+          minimum: 1
+          type: integer
+      responses:
+        '200':
+          content:
+            application/json:
+              schema:
+                type: object
+                allOf:
+                - $ref: '#/components/schemas/QueryResponse'
+                properties:
+                  results:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/Diagram'
+          description: OK
+      tags:
+        - Diagram
+    """
+    require(request.authz.logged_in)
+    parser = QueryParser(request.args, request.authz)
+    collection_id = first(parser.filters.get('collection_id'))
+    if collection_id:
+        get_db_collection(collection_id)
+        q = Diagram.by_collection(collection_id).filter(Diagram.role_id == request.authz.id)  # noqa
+    else:
```

Comment:
That seems like a false A or B to me. Can people either see their own diagrams, or those in a collection? I think this might be cleaner if we prepare a SQLA q object that filters for the user's collections, and then drill it down with collection and role filters.

Reply:
done